lib/resty/upload.lua

```diff
@@ -62,20 +65,35 @@ function _M.new(self, chunk_size, max_line_size)
         return nil, err
     end
 
+    if restore_body_buffer then
+        ngx_init_body(chunk_size)
+    end
+
     local read2boundary, err = sock:receiveuntil("--" .. boundary)
     if not read2boundary then
+
+        if restore_body_buffer then
+            ngx_finish_body(self)
+        end
+
         return nil, err
     end
 
     local read_line, err = sock:receiveuntil("\r\n")
     if not read_line then
+
+        if restore_body_buffer then
+            ngx_finish_body(self)
+        end
+
         return nil, err
     end
 
     return setmetatable({
         sock = sock,
         size = chunk_size or CHUNK_SIZE,
         line_size = max_line_size or MAX_LINE_SIZE,
+        resore_body = restore_body_buffer or false,
```

Comment:
`resore` -> `restore`? Also `or false` not needed, though, doesn't hurt much either.